packages/core/src/particle/modules/shape/BoxShape.ts

```diff
@@ -28,4 +47,26 @@ export class BoxShape extends BaseShape {
     ShapeUtils._randomPointUnitSphere(direction, rand);
     Vector3.lerp(defaultDirection, direction, this.randomDirectionAmount, direction);
   }
+
+  /**
+   * @internal
+   */
+  override _getDirectionRange(min: Vector3, max: Vector3) {
+    const { randomDirectionAmount } = this;
+    min.set(-randomDirectionAmount, -randomDirectionAmount, -1);
+    max.set(randomDirectionAmount, randomDirectionAmount, randomDirectionAmount);
+  }
```

Comment:
Should use `randomDirectionAmount*Math.PI*2` and get the angle from (0,0,-1)！Then you can compute edge vector!